File: public/sdk/java/ariana_enterprise_java/src/main/java/br/com/arianamoveis/enterprise/ArianaEnterpriseClient.java
package br.com.arianamoveis.enterprise;

import java.io.IOException;
import java.net.URI;
import java.net.http.*;
import java.time.Duration;
import java.util.*;

public class ArianaEnterpriseClient {
    private final String apiKey;
    private final String bearerToken;
    private final String baseUrl;
    private final HttpClient http;

    private ArianaEnterpriseClient(Builder builder) {
        this.apiKey = builder.apiKey;
        this.bearerToken = builder.bearerToken;
        String env = builder.environment == null ? "sandbox" : builder.environment;
        this.baseUrl = builder.baseUrl != null ? builder.baseUrl :
            ("production".equalsIgnoreCase(env) ? "https://ariana-backend.onrender.com/api/v1/enterprise" : "https://ariana-backend.onrender.com/api/v1/enterprise");
        this.http = HttpClient.newBuilder().connectTimeout(Duration.ofMillis(builder.timeoutMs)).build();
    }

    public static Builder builder() { return new Builder(); }

    public Map<String, Object> health() throws IOException, InterruptedException {
        return request("GET", "/health", null);
    }

    public Catalog catalog() { return new Catalog(this); }
    public Products products() { return new Products(this); }
    public Orders orders() { return new Orders(this); }
    public Invoice invoice() { return new Invoice(this); }
    public Tracking tracking() { return new Tracking(this); }
    public Webhooks webhooks() { return new Webhooks(this); }

    Map<String, Object> request(String method, String path, String jsonBody) throws IOException, InterruptedException {
        HttpRequest.Builder req = HttpRequest.newBuilder()
            .uri(URI.create(baseUrl + path))
            .timeout(Duration.ofSeconds(30))
            .header("Content-Type", "application/json")
            .header("Accept", "application/json");

        if (bearerToken != null && !bearerToken.isBlank()) req.header("Authorization", "Bearer " + bearerToken);
        else if (apiKey != null && !apiKey.isBlank()) req.header("x-ariana-key", apiKey);

        if ("GET".equalsIgnoreCase(method)) req.GET();
        else req.method(method, HttpRequest.BodyPublishers.ofString(jsonBody == null ? "{}" : jsonBody));

        HttpResponse<String> response = http.send(req.build(), HttpResponse.BodyHandlers.ofString());
        Map<String, Object> result = new LinkedHashMap<>();
        result.put("status", response.statusCode());
        result.put("body", response.body());
        result.put("ok", response.statusCode() >= 200 && response.statusCode() < 300);
        if (response.statusCode() == 429) result.put("retryAfter", response.headers().firstValue("Retry-After").orElse(""));
        return result;
    }

    public static class Builder {
        private String apiKey;
        private String bearerToken;
        private String environment = "sandbox";
        private String baseUrl;
        private int timeoutMs = 30000;

        public Builder apiKey(String apiKey) { this.apiKey = apiKey; return this; }
        public Builder bearerToken(String bearerToken) { this.bearerToken = bearerToken; return this; }
        public Builder environment(String environment) { this.environment = environment; return this; }
        public Builder baseUrl(String baseUrl) { this.baseUrl = baseUrl; return this; }
        public Builder timeoutMs(int timeoutMs) { this.timeoutMs = timeoutMs; return this; }
        public ArianaEnterpriseClient build() { return new ArianaEnterpriseClient(this); }
    }
}

File: public/sdk/dotnet/ariana_enterprise_dotnet/src/ArianaEnterpriseClient.cs
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;

namespace ArianaEnterprise;

public sealed class ArianaEnterpriseOptions
{
    public string ApiKey { get; set; } = "";
    public string Environment { get; set; } = "sandbox";
    public string BaseUrl { get; set; } = "https://ariana-backend.onrender.com/api/v1/enterprise";
    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(30);
}

public sealed class ArianaProduct
{
    public string Sku { get; set; } = "";
    public string Name { get; set; } = "";
    public decimal Price { get; set; }
    public int Stock { get; set; }
}

public sealed class ArianaEnterpriseClient
{
    private readonly HttpClient _http;
    private readonly ArianaEnterpriseOptions _options;
    private static readonly JsonSerializerOptions JsonOptions = new(JsonSerializerDefaults.Web);

    public CatalogClient Catalog { get; }
    public ProductsClient Products { get; }
    public OrdersClient Orders { get; }

    public ArianaEnterpriseClient(ArianaEnterpriseOptions options, HttpClient? httpClient = null)
    {
        _options = options;
        _http = httpClient ?? new HttpClient();
        _http.Timeout = options.Timeout;
        _http.DefaultRequestHeaders.Add("x-ariana-key", options.ApiKey);
        Catalog = new CatalogClient(this);
        Products = new ProductsClient(this);
        Orders = new OrdersClient(this);
    }

    public Task<JsonDocument> HealthAsync() => SendAsync(HttpMethod.Get, "/health");

    internal async Task<JsonDocument> SendAsync(HttpMethod method, string path, object? body = null)
    {
        var request = new HttpRequestMessage(method, _options.BaseUrl.TrimEnd('/') + path);
        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
        if (body is not null)
        {
            request.Content = new StringContent(JsonSerializer.Serialize(body, JsonOptions), Encoding.UTF8, "application/json");
        }

        var response = await _http.SendAsync(request);
        var text = await response.Content.ReadAsStringAsync();
        if (!response.IsSuccessStatusCode)
        {
            throw new ArianaEnterpriseException((int)response.StatusCode, text);
        }
        return JsonDocument.Parse(string.IsNullOrWhiteSpace(text) ? "{}" : text);
    }
}

public sealed class CatalogClient
{
    private readonly ArianaEnterpriseClient _client;
    internal CatalogClient(ArianaEnterpriseClient client) => _client = client;
    public Task<JsonDocument> PushAsync(IEnumerable<ArianaProduct> products, string manufacturer = "ariana_moveis") =>
        _client.SendAsync(HttpMethod.Post, "/catalog/push", new { manufacturer, products });
}

public sealed class ProductsClient
{
    private readonly ArianaEnterpriseClient _client;
    internal ProductsClient(ArianaEnterpriseClient client) => _client = client;
    public Task<JsonDocument> UpdateStockAsync(string sku, int stock) =>
        _client.SendAsync(HttpMethod.Put, $"/products/{sku}/stock", new { stock });
    public Task<JsonDocument> UpdatePriceAsync(string sku, decimal price) =>
        _client.SendAsync(HttpMethod.Put, $"/products/{sku}/price", new { price });
}

public sealed class OrdersClient
{
    private readonly ArianaEnterpriseClient _client;
    internal OrdersClient(ArianaEnterpriseClient client) => _client = client;
    public Task<JsonDocument> CreateAsync(object order) => _client.SendAsync(HttpMethod.Post, "/orders", order);
}

public sealed class ArianaEnterpriseException : Exception
{
    public int StatusCode { get; }
    public string ResponseBody { get; }
    public ArianaEnterpriseException(int statusCode, string responseBody) : base($"Ariana Enterprise HTTP {statusCode}")
    {
        StatusCode = statusCode;
        ResponseBody = responseBody;
    }
}
